component/core/src/main/java/org/exoplatform/social/core/binding/listener/SpaceBindingMembershipGroupEventListener.java

```diff
@@ -0,0 +1,129 @@
+  public void postSave(Membership m, boolean isNew) throws Exception {
+    String userName = m.getUserName();
+    String groupId = m.getGroupId();
+    if (isNew && !isASpaceGroup(groupId)) {
+      RequestLifeCycle.begin(PortalContainer.getInstance());
+      try {
+        if (isUserNewMemberToGroup(userName, groupId)) {
+          groupSpaceBindingService = CommonsUtils.getService(GroupSpaceBindingService.class);
+          spaceService = CommonsUtils.getService(SpaceService.class);
+
+          // Retrieve all bindings of the group.
+          List<GroupSpaceBinding> groupSpaceBindings = groupSpaceBindingService.findGroupSpaceBindingsByGroup(groupId);
+          // For each bound space of the group add a user binding to it.
+          for (GroupSpaceBinding groupSpaceBinding : groupSpaceBindings) {
+            Space space = spaceService.getSpaceById(groupSpaceBinding.getSpaceId());
+            groupSpaceBindingService.saveUserBinding(userName, groupSpaceBinding, space);
+          }
+        }
+      } catch (Exception e) {
+        LOG.warn("Problem occurred when saving user bindings for user ({}) from group ({}): ", userName, groupId, e);
+      } finally {
+        RequestLifeCycle.end();
+      }
+    }
+  }
+
+  @Override
+  public void postDelete(Membership m) throws Exception {
+    String userName = m.getUserName();
+    String groupId = m.getGroupId();
+    if (!isASpaceGroup(groupId)) {
+      RequestLifeCycle.begin(PortalContainer.getInstance());
+      try {
+        if (isUserNoMoreMemberOfGroup(userName, groupId)) {
+          groupSpaceBindingService = CommonsUtils.getService(GroupSpaceBindingService.class);
+          spaceService = CommonsUtils.getService(SpaceService.class);
+          // Retrieve removed user's all bindings.
+          List<UserSpaceBinding> userSpaceBindings = groupSpaceBindingService.findUserBindingsByGroup(groupId, userName);
+          // Remove them.
+          for (UserSpaceBinding userSpaceBinding : userSpaceBindings) {
+            // Delete user binding.
+            groupSpaceBindingService.deleteUserSpaceBinding(userSpaceBinding);
+            // Check if user has other bindings to the space.
+            boolean hasOtherBindings = groupSpaceBindingService
+                                                               .findUserSpaceBindingsBySpace(userSpaceBinding.getGroupBinding()
+                                                                                                             .getSpaceId(),
+                                                                                             userSpaceBinding.getUser())
+                                                               .size() > 0;
+            if (!hasOtherBindings) {
+              Space space = spaceService.getSpaceById(userSpaceBinding.getGroupBinding().getSpaceId());
+              // Remove user membership from the space.
+              spaceService.removeMember(space, userName);
+            }
+          }
```

Comment:
Why dont you call
groupSpaceBindingService.deleteUserBindingAndSpaceMembership(userSpaceBinding) ?
This function make all tests to check if the user have other bindings and if he is memberbefore

```suggestion
          for (UserSpaceBinding userSpaceBinding : userSpaceBindings) {
            // Delete user binding.
            groupSpaceBindingService.deleteUserBindingAndSpaceMembership(userSpaceBinding);
          }
```

Reply:
> Why dont you call
> groupSpaceBindingService.deleteUserBindingAndSpaceMembership(userSpaceBinding) ?
> This function make all tests to check if the user have other bindings and if he is memberbefore

It was made that way but according to the functional specification: If a user is removed from a group while this group has a bindings, he is deleted automatically from the bound spaces if not already bound with other groups to these space, it is not the same process.